### PR TITLE
chore: fix package specs #282

### PR DESCRIPTION
**Pull request by atarashansky**
#282 

Tests were failing. There was a dependency conflict with webpack=5.34.0. 

Downgrading webpack to 4.0.0 (and associated packages) fixes the issue. The tests are now failing for actual test reasons, the rest of this PR will be focused on cleaning those up.

### CODE REVIEW

**Comment by atarashansky** [comment]
Tests complete!

**Review by seve — COMMENTED** [review]
I believe the original reason for the webpack version bump was for vulnerability reasons, can you confirm that we aren't downgrading to any vulnerable packages

File: client/src/reducers/undoableConfig.ts [hunk 1]
```diff
@@ -261,10 +261,7 @@ return true if objA and objB are ===, OR if:
   - have same own properties
   - all values are strict equal (===)
 */
-function shallowObjectEq(
-  objA: Record<string | number | symbol, unknown>,
-  objB: Record<string | number | symbol, unknown>
-) {
+function shallowObjectEq(objA: AnyAction, objB: AnyAction) {
```
[the remark this answers is not in the document]
Comment (atarashansky) [reply client/src/reducers/undoableConfig.ts:264]: Typescript was complaining. `shallowObjectEq` was being passed something of type `AnyAction`, so I changed `objA` typing to `AnyAction`

**Comment by atarashansky** [comment]
> I believe the original reason for the webpack version bump was for vulnerability reasons, can you confirm that we aren't downgrading to any vulnerable packages

Is this something I would do with `npm audit`?

**Comment by atarashansky** [comment]
The problem is that there is no version of `optimize-css-assets-webpack-plugin` that works with `webpack^5.0.0`. That was the cause of the error logs in #282.

**Comment by atarashansky** [comment]
Relevant thread: https://github.com/NMFR/optimize-css-assets-webpack-plugin/issues/150

**Comment by seve** [comment]
Should we instead migrate to https://github.com/webpack-contrib/css-minimizer-webpack-plugin

**Comment by atarashansky** [comment]
> Should we instead migrate to webpack-contrib/css-minimizer-webpack-plugin

I can give that a try.

**Review by seve — APPROVED** [review]

**Pull request merged**